### PR TITLE
Handle unused errors

### PR DESCRIPTION
**Pull request by joshblum**
https://staticcheck.io/docs/ seems like a useful tool and may be worth integrating into CI

ran `$ staticcheck -checks "SA4006" ./...  | grep "this value of err"` to handle unused errors.

### CODE REVIEW

**Comment by joshblum** [comment]
@jzila 

```
rn-story-mode........................................(no files to check)Skipped
eslint...............................................(no files to check)Skipped
tsc..................................................(no files to check)Skipped
prettier.............................................(no files to check)Skipped
go fmt...................................................................Passed
go vet...................................................................Passed
go lint..................................................................Failed
hookid: go-lint

go/kbfs/simplefs/simplefs_test.go:137:1: context.Context should be the first parameter of a function
go/kbfs/simplefs/simplefs_test.go:174:1: context.Context should be the first parameter of a function

```

**Comment by strib** [comment]
@joshblum not sure what tool this is, but we explicitly mask out that golint ctx check for tests in https://github.com/keybase/client/blob/a19fd3247f50b28058dbcd6ecc56616a26cf09d7/go/Makefile#L36

I don't mind changing the code this way, but presumably there will be a ton of other lint failures for the other errors we filter out.

**Comment by joshblum** [comment]
@strib this just came from the precommit hook not the tool, could revert the changes and skip the precommit checks if it will pass CI

File: go/kbfs/libkbfs/tlf_journal_test.go [hunk 1]
```diff
@@ -532,6 +532,7 @@ func testTLFJournalBlockOpDiskByteLimit(t *testing.T, ver kbfsmd.MetadataVer) {
 	md := config.makeMD(kbfsmd.RevisionInitial, kbfsmd.ID{})
 	err = tlfJournal.doOnMDFlushAndRemoveFlushedMDEntry(
 		ctx, kbfsmd.ID{}, &RootMetadataSigned{RootMetadataSigned: kbfsmd.RootMetadataSigned{MD: md.bareMd}})
+	require.NoError(t, err)
```
Comment (joshblum) [inline go/kbfs/libkbfs/tlf_journal_test.go]: @strib is an error expected here? https://ci.keyba.se/job/client/job/PR-17681/4/execution/node/2354/log/

Comment (strib) [reply go/kbfs/libkbfs/tlf_journal_test.go]: I took a quick look and yeah it seems like an error is expected here, it's doing something kinda funny for test purposes.

Comment (joshblum) [reply go/kbfs/libkbfs/tlf_journal_test.go]: thx!

**Comment by joshblum** [comment]
@jzila ptal

File: go/engine/passphrase_recover.go [hunk 2]
```diff
@@ -251,6 +251,9 @@ func (e *PassphraseRecover) changePassword(mctx libkb.MetaContext) (err error) {
 	// We either have no server keys or the user is OK with resetting them
 	// Prompt the user for a new passphrase.
 	passphrase, err := e.promptPassphrase(mctx)
+	if err != nil {
+		return err
+	}
```
Comment (jzila) [inline go/engine/passphrase_recover.go]: Was this a bug?

Comment (joshblum) [reply go/engine/passphrase_recover.go]: yes i don't think i the error was intentionally ignored

**Pull request merged**